arviz/tests/test_rcparams.py

```diff
@@ -135,6 +137,42 @@ def test_make_validate_choice(args, allow_none, typeof):
         assert value in accepted_values or value is None
 
 
+@pytest.mark.parametrize("allow_none", (True, False))
+@pytest.mark.parametrize("allow_auto", (True, False))
+@pytest.mark.parametrize("length", (2, 3))
+@pytest.mark.parametrize(
+    "args",
+    [
+        (False, (1, 2)),
+        ("Only ordered iterable", set(["a", "b", "c"])),
+        (False, None),
+        (False, "auto"),
+    ],
+)
+def test_make_iterable_validator(args, length, allow_auto, allow_none):
+    scalar_validator = _validate_float_or_none
+    validate_iterable = make_iterable_validator(
+        scalar_validator, length=length, allow_none=allow_none, allow_auto=allow_auto
+    )
+    raise_error, value = args
+    if (
+        value not in {None, "auto"}
+        and not isinstance(value, (set, frozenset))
+        and len(value) != length
+    ):
+        raise_error = "Iterable must be of length"
+    if value is None and not allow_none:
+        raise_error = "Only ordered iterable"
+    if value == "auto" and not allow_auto:
+        raise_error = "Could not convert"
```

Comment:
I was quite puzzled by this, I expected the only ordered iterable error in this case, turns out `np.iterable` returns true for strings, therefore, `"14"` would be accepted as `(1, 4)`.

Reply:
Thanks, didn't think of that!

Reply:
Actually ```"14"``` after conversion is stored in value as ```('14', )``` and I don't think we need iterables of length one anyway so should I change ```if np.iterable(value) and not isinstance(value, (set, frozenset)):``` to ```if np.iterable(value) and len(val) > 1 and not isinstance(value, (set, frozenset)):```? And we won't need to exclude strings here since ```"(1, 4)"``` after conversion to iterable will automatically be evaluated by ```scalar_validator```

Reply:
Ohh, true, my bad. as `tuple` is called, strings like `"auto"` or `"14"` are converted to a one element tuple, it is not possible for value to still be of type string at this point of the function. No need to modify the code (in a general case where length is None a 1 item iterable should be accepted)